src/librc/librc.c

```diff
@@ -755,8 +755,10 @@ rc_service_mark(const char *service, const RC_SERVICE state)
 					    RC_SVCDIR "/%s/%s",
 					    rc_parse_service_state(RC_SERVICE_WASINACTIVE),
 					    base);
-					if (symlink(init, was) == -1)
+					if (symlink(init, was) == -1) {
+						free(init);
```

Comment:
this looks correct, but from scanning the rest of this func, seems like there's other places this leaks that need fixing too ?

Reply:
sorry, I can't spot it - could you point one out?

Reply:
sorry, i was looking at the file w/out your patch applied, and thought it was fixing an earlier line

Reply:
np